lib/Backend/amd64/LowererMDArch.cpp

```diff
@@ -1486,7 +1486,14 @@ LowererMDArch::LowerEntryInstr(IR::EntryInstr * entryInstr)
     if (this->m_func->HasArgumentSlot() && (this->m_func->IsStackArgsEnabled() ||
         this->m_func->IsJitInDebugMode() ||
         // disabling apply inlining leads to explicit load from the zero-inited slot
-        this->m_func->GetJnFunction()->IsInlineApplyDisabled()))
+        this->m_func->GetJnFunction()->IsInlineApplyDisabled())
+#ifdef BAILOUT_INJECTION
+        || Js::Configuration::Global.flags.IsEnabled(Js::BailOutFlag)
+        || Js::Configuration::Global.flags.IsEnabled(Js::BailOutAtEveryLineFlag)
+        || Js::Configuration::Global.flags.IsEnabled(Js::BailOutAtEveryByteCodeFlag)
+        || Js::Configuration::Global.flags.IsEnabled(Js::BailOutByteCodeFlag)
```

Comment:
Looks like BAILOUT_INJECTION  is enabled only in DEBUG build. Since most Func's have hasBailout flag set, we will differ a lot in eliminating zero-init between debug and test builds if we have hasBailout check instead of explicit flag check ?